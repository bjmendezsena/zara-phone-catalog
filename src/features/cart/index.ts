export * from "./context";
export * from "./helpers";
export * from "./types";
export * from "./components";
