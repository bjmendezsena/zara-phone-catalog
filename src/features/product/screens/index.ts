export * from "./ProductDetails";
export * from "./ProductGrid";

