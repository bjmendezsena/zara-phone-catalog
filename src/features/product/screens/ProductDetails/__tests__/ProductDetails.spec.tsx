import {
  getHtml,
  renderWithProviders,
  mockProductDetailsResponse,
  waitFor,
} from "@/tests";
import { ProductDetails, ProductDetailsProps } from "@/features/product";

describe(`<${ProductDetails.name}/>`, () => {
  const defaultProps: ProductDetailsProps = {
    productId: mockProductDetailsResponse.id,
  };
  const getComponent = (props: Partial<ProductDetailsProps> = {}) =>
    renderWithProviders(<ProductDetails {...defaultProps} {...props} />);

  it("renders without crashing", async () => {
    const { container } = getComponent();
    await waitFor(() => {
      const htmlString = getHtml(container);
      expect(htmlString).toMatchSnapshot();
      expect(htmlString.toString()).toContain(mockProductDetailsResponse.name);
      expect(htmlString.toString()).toContain(mockProductDetailsResponse.brand);
      expect(htmlString.toString()).toContain(
        mockProductDetailsResponse.description
      );
    });
  });
  it("renders with loading", async () => {
    const { getByText } = renderWithProviders(
      <ProductDetails {...defaultProps} />
    );
    expect(getByText("Loading...")).toBeInTheDocument();
  });
});
