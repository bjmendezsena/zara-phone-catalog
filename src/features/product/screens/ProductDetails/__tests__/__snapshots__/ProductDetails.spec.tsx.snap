// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ProductDetails/> renders without crashing 1`] = `
"<div>
  <div
    class="flex items-center justify-center min-h-screen"
  >
    <span
      class="text-secondary"
    >
      Loading...
    </span>
  </div>
</div>"
`;

exports[`<ProductDetails/> renders without crashing 2`] = `
"<div>
  <div
    class="flex flex-col py-4 gap-8 w-full"
  >
    <div
      class="mb-8 px-4"
    >
      <button
        class="flex items-center space-x-2 cursor-pointer"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chevron-left"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="0.75"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m15 18-6-6 6-6"
          />
        </svg>
        <span
          class="text-sm uppercase tracking-wide"
        >
          BACK
        </span>
      </button>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-16 xl:px-[360px]"
    >
      <div
        class="flex justify-center items-center"
      >
        <div
          class="w-[400px] h-[500px] flex items-center justify-center"
        >
          <img
            alt="Google Pixel 8a"
            class="w-full h-full object-contain"
            data-priority="true"
            height="500"
            src="http://prueba-tecnica-api-tienda-moviles.onrender.com/images/GPX-8A-obsidiana.webp"
            width="400"
          />
        </div>
      </div>
      <div
        class="space-y-8 px-4 md:px-14 lg:px-4"
      >
        <div>
          <h1
            class="text-3xl font-light text-primary mb-2 uppercase tracking-wide"
          >
            Pixel 8a
          </h1>
          <p
            class="text-lg"
          >
            From 459.00 EUR
          </p>
        </div>
        <div>
          <h3
            class="text-sm  mb-4 uppercase tracking-wide"
          >
            STORAGE 
            <span
              class="uppercase"
            >
              How much space do you need?
            </span>
          </h3>
          <div
            class="flex divide-x-4"
          >
            <button
              aria-label="Select 128 GB storage"
              class="cursor-pointer px-6 py-3 border text-sm transition-colors border-gray-300 border-r-transparent text-secondary hover:border-primary hover:border-r"
              data-testid="storage-option-128 GB"
            >
              128 GB
            </button>
            <button
              aria-label="Select 256 GB storage"
              class="cursor-pointer px-6 py-3 border text-sm transition-colors border-gray-300 text-secondary hover:border-primary hover:border-r border-r-gray-300"
              data-testid="storage-option-256 GB"
            >
              256 GB
            </button>
          </div>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <h3
            class="text-sm mb-4 uppercase tracking-wide"
          >
            COLOR. 
            <span
              class="uppercase"
            >
              Pick your favourite
            </span>
          </h3>
          <div
            class="flex gap-3"
          >
            <button
              aria-label="Select Obsidiana color"
              class="w-[24px] h-[24px] border-2 transition-all cursor-pointer p-[2px] border-gray-300 hover:border-primary"
              data-testid="color-option-Obsidiana"
            >
              <div
                aria-hidden="true"
                class="w-full h-full"
                style="background-color: rgb(0, 0, 0);"
              />
            </button>
            <button
              aria-label="Select Porcelana color"
              class="w-[24px] h-[24px] border-2 transition-all cursor-pointer p-[2px] border-gray-300 hover:border-primary"
              data-testid="color-option-Porcelana"
            >
              <div
                aria-hidden="true"
                class="w-full h-full"
                style="background-color: rgb(245, 245, 245);"
              />
            </button>
            <button
              aria-label="Select Celeste color"
              class="w-[24px] h-[24px] border-2 transition-all cursor-pointer p-[2px] border-gray-300 hover:border-primary"
              data-testid="color-option-Celeste"
            >
              <div
                aria-hidden="true"
                class="w-full h-full"
                style="background-color: rgb(135, 206, 235);"
              />
            </button>
          </div>
          <span
            class="text-xs uppercase tracking-wide min-h-[15px]"
          />
        </div>
        <div
          class="fixed border-none bottom-0 left-0 right-0 z-50 px-[16px] py-[20px] md:p-0 bg-white md:relative md:bottom-auto md:left-auto md:right-auto md:z-auto"
        >
          <button
            class="w-full py-4 text-sm uppercase tracking-wide transition-colors mx-0 sm:mx-auto rounded-none sm:rounded bg-disabled text-text-disabled cursor-not-allowed"
            disabled=""
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
    <div
      class="mt-4 xl:px-[360px] px-4"
    >
      <h2
        class="text-xl font-light text-primary mb-8 uppercase tracking-wide"
      >
        SPECIFICATIONS
      </h2>
      <div
        class="space-y-1"
      >
        <div
          class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
        >
          <div
            class="text-sm uppercase tracking-wide font-medium"
          >
            BRAND
          </div>
          <div
            class="text-sm"
          >
            Google
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
        >
          <div
            class="text-sm uppercase tracking-wide font-medium"
          >
            NAME
          </div>
          <div
            class="text-sm"
          >
            Pixel 8a
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
        >
          <div
            class="text-sm uppercase tracking-wide font-medium"
          >
            DESCRIPTION
          </div>
          <div
            class="text-sm"
          >
            Descubre Pixel 8a, creado por Google. Saca fotos magníficas con la Cámara Pixel. Haz más en menos tiempo con la IA de Google, como arreglar imágenes borrosas, filtrar llamadas y aprender cosas nuevas. Sus funciones de seguridad del más alto nivel ayudan a proteger tus datos. Y se ha diseñado para durar. Todo, a un precio excepcional.
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
        >
          <div
            class="text-sm uppercase tracking-wide font-medium"
          >
            SCREEN
          </div>
          <div
            class="text-sm"
          >
            6.1" OLED Actua
          </div>
        </div>
        <div
          class="grid grid-col..."
`;
